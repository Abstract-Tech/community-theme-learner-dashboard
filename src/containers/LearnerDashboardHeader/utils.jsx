import { getConfig } from '@edx/frontend-platform';
import { getAuthenticatedUser } from '@edx/frontend-platform/auth';

export const isEnterpriseUser = () => {
  const authenticatedUser = getAuthenticatedUser();
  const userRoleNames = authenticatedUser ? authenticatedUser.roles.map(role => role.split(':')[0]) : [];

  return userRoleNames.includes('enterprise_learner');
};

export const showOrdersAndSubscriptionsMenuItem = () => (getConfig().ENABLE_B2C_SUBSCRIPTIONS?.toLowerCase() === 'true' && !isEnterpriseUser());

export default isEnterpriseUser;
