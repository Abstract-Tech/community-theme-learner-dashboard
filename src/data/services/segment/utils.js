/* eslint-disable import/prefer-default-export */
import { sendTrackEvent } from '@edx/frontend-platform/analytics';
import { appName } from './constants';

export const trackEvent = (name, options = {}) => sendTrackEvent(
  name,
  { ...options, app_name: appName },
);
