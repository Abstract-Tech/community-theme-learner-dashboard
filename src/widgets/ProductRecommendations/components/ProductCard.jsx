import React from 'react';
import PropTypes from 'prop-types';
import {
  Badge,
  Card,
  Truncate,
  Hyperlink,
} from '@edx/paragon';

const ProductCard = ({
  title,
  subtitle,
  headerImage,
  schoolLogo,
  courseType,
  url,
}) => (
  <div className="base-card-wrapper">
    <Hyperlink destination={url}>
      <Card className="base-card light" variant="light">
        <Card.ImageCap
          src={headerImage}
          srcAlt={`header image for ${title}`}
          logoSrc={schoolLogo}
          logoAlt={`logo for ${subtitle}`}
        />
        <Card.Header
          className="mt-2"
          title={(
            <Truncate lines={3} ellipsis="â€¦" className="product-card-title">
              {title}
            </Truncate>
          )}
          subtitle={(
            <Truncate lines={1} className="product-card-subtitle">
              {subtitle}
            </Truncate>
          )}
        />
        <Card.Section>
          <div className="product-badge">
            <Badge>{courseType}</Badge>
          </div>
        </Card.Section>
      </Card>
    </Hyperlink>
  </div>
);

ProductCard.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  headerImage: PropTypes.string.isRequired,
  schoolLogo: PropTypes.string.isRequired,
  courseType: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default ProductCard;
