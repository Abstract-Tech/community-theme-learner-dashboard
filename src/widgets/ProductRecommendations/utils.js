import PropTypes from 'prop-types';

export const courseShape = {
  uuid: PropTypes.string,
  title: PropTypes.string,
  image: PropTypes.shape({
    src: PropTypes.string,
  }),
  prospectusPath: PropTypes.string,
  owners: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string,
      name: PropTypes.string,
      logoImageUrl: PropTypes.string,
    }),
  ),
  activeCourseRun: PropTypes.shape({
    key: PropTypes.string,
    marketingUrl: PropTypes.string,
  }),
  courseType: PropTypes.string,
};

export const courseTypeToProductTypeMap = {
  course: 'Course',
  'verified-audit': 'Course',
  verified: 'Course',
  audit: 'Course',
  'credit-verified-audit': 'Course',
  'spoc-verified-audit': 'Course',
  professional: 'Professional Certificate',
  'bootcamp-2u': 'Boot Camp',
  'executive-education-2u': 'Executive Education',
  'executive-education': 'Executive Education',
  masters: "Master's",
  'masters-verified-audit': "Master's",
};

export const wait = (time) => new Promise((resolve) => {
  setTimeout(resolve, time);
});
