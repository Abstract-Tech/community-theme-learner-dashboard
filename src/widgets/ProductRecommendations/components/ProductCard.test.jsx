import React from 'react';
import { shallow } from 'enzyme';

import { mockCrossProductCourses } from '../testData';
import ProductCard from './ProductCard';
import { courseTypeToProductTypeMap } from '../utils';

describe('ProductRecommendations ProductCard', () => {
  const course = mockCrossProductCourses[0];
  const {
    title,
    owners: [{ name: subtitle }],
    image: { src: headerImage },
    owners: [{ logoImageUrl: schoolLogo }],
  } = course;

  const props = {
    title,
    subtitle,
    headerImage,
    schoolLogo,
    courseType: courseTypeToProductTypeMap[course.courseType],
    url: `${course.marketingUrl}&linked_from=recommender`,
  };

  it('matches snapshot', () => {
    expect(shallow(<ProductCard {...props} />)).toMatchSnapshot();
  });
});
