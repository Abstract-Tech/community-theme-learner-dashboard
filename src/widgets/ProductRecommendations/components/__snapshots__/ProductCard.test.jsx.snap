// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRecommendations ProductCard matches snapshot 1`] = `
<div
  className="base-card-wrapper"
>
  <Hyperlink
    destination="https://www.edx.org/course/introduction-to-computer-sceince"
  >
    <Card
      className="base-card light"
      variant="light"
    >
      <Card.ImageCap
        logoAlt="logo for Harvard University"
        logoSrc="http://www.image.com/ef72daf3-c9a1-4c00-ba37-b3514392bdcf-8839c516815a.png"
        src="https://www.image-2.com/ed79a49b-64c1-48d2-afdc-054bf921e38d-6a76ceb47dea.small.jpg"
        srcAlt="header image for Introduction to Computer Science"
      />
      <Card.Header
        className="mt-2"
        subtitle={
          <Truncate
            className="product-card-subtitle"
            lines={1}
          >
            Harvard University
          </Truncate>
        }
        title={
          <Truncate
            className="product-card-title"
            ellipsis="â€¦"
            lines={3}
          >
            Introduction to Computer Science
          </Truncate>
        }
      />
      <Card.Section>
        <div
          className="product-badge"
        >
          <Badge>
            Executive Education
          </Badge>
        </div>
      </Card.Section>
    </Card>
  </Hyperlink>
</div>
`;
